packages/react/ds/src/card/card.tsx

```diff
@@ -1,61 +1,136 @@
-import React, { cloneElement, ReactElement } from 'react';
+import { useMemo } from 'react';
+import { Button } from '../button/button.js';
+import { ButtonProps } from '../button/types.js';
 import { Heading } from '../heading/heading.js';
 import { Icon, IconPropTypes } from '../icon/icon.js';
-import { Link } from '../link/link.js';
+import { Link, LinkProps } from '../link/link.js';
 import { Paragraph } from '../paragraph/paragraph.js';
+import { Tag, TagProps } from '../tag/tag.js';
 
-type Action = {
-  href: string;
-  text: string;
-};
+type Action =
+  | (ButtonProps & { type: 'button' })
+  | (LinkProps & { type: 'link' });
 
 export type CardProps = {
   type: 'vertical' | 'horizontal';
+  inset?: 'body' | 'full' | 'none';
   title?: string;
+  subTitle?: string;
   href?: string;
   img?: string;
+  tag?: TagProps;
   icon?: IconPropTypes;
   content?: string;
-  actions?: Action[];
+  action?: Action;
 };
 
-export const Card = (props: CardProps) => {
-  const { type = 'vertical', title, img, icon, content, actions, href } = props;
-  const layoutClasses = {
-    vertical: 'gi-card gi-card-vertical',
-    horizontal: 'gi-card gi-card-horizontal',
-  };
+// Memoized truncation utility
+const useTruncatedText = (text: string | undefined, maxLength: number) => {
```

Comment:
use [text overflow](https://tailwindcss.com/docs/text-overflow) instead of this function.